### PR TITLE
Documentation fixes for Samza SQL

### PR DESCRIPTION
**Pull request by shenodaguirguis**

### CODE REVIEW

File: docs/startup/quick-start/versioned/samza.md [hunk 1]
```diff
@@ -0,0 +1,245 @@
+---
+layout: page
+title: Samza Quick Start
+---
+<!--
+   Licensed to the Apache Software Foundation (ASF) under one or more
```
Comment (vjagadish1989) [inline docs/startup/quick-start/versioned/samza.md]: I'm assuming this is just a rename?

File: docs/learn/documentation/versioned/api/samza-sql.md [hunk 2]
```diff
@@ -19,65 +19,57 @@ title: Samza SQL
...
+### Code Examples
 
-### Running Samza SQL on your local machine
-The [Samza SQL console](https://samza.apache.org/learn/tutorials/0.14/samza-tools.html) allows you to experiment with Samza SQL locally on your machine. 
+The [Hello Samza](https://github.com/apache/samza-hello-samza/tree/master/src/main/java/samza/examples/sql) 
+SQL examples demonstrate how to use Samza SQL API.  
 
-#### Setup Kafka
-Follow the instructions from the [Kafka quickstart](http://kafka.apache.org/quickstart) to start the zookeeper and Kafka server.
+- The [sampling](https://github.com/apache/samza-hello-samza/tree/master/src/main/java/samza/examples/sql/samza-sql-samples-yarn)
```
Comment (vjagadish1989) [inline docs/learn/documentation/versioned/api/samza-sql.md]: nit: link appears broken.. Do you have a pending PR for the SQL examples on YARN? 

if so, maybe worth getting that committed at first?

File: docs/learn/documentation/versioned/api/samza-sql.md [hunk 3]
```diff
@@ -19,65 +19,57 @@ title: Samza SQL
...
-```bash
-> ./scripts/samza-sql-console.sh --sql "insert into log.consoleoutput select Name, OldCompany, NewCompany from kafka.ProfileChangeStream"
-```
+Support for SQL internally uses Apache Calcite, which provides 
+SQL parsing and query planning. The query is automatically translated 
+to Samza's high level API and runs on Samza's execution engine.
 
-You can also filter messages in the input stream based on some predicate. In this example, we filter profiles currently working at LinkedIn, whose previous employer matches the regex `.*soft`. The function `RegexMatch(regex, company)` is an example of 
-a UDF that defines a predicate. 
+The mapping from SQL to the Samza's high level API is a simple
```
Comment (vjagadish1989) [inline docs/learn/documentation/versioned/api/samza-sql.md]: worth having a table of supported operations. what do you think?

https://docs.confluent.io/current/ksql/docs/developer-guide/join-streams-and-tables.html

File: docs/startup/quick-start/versioned/samza.md (hunk 1 above)
Comment (shenodaguirguis) [reply docs/startup/quick-start/versioned/samza.md]: yes

File: docs/learn/documentation/versioned/api/samza-sql.md (hunk 2 above)
Comment (shenodaguirguis) [reply docs/learn/documentation/versioned/api/samza-sql.md]: indeed & agreed. The examples PR is https://github.com/apache/samza-hello-samza/pull/46

Comment (atoomula) [reply docs/learn/documentation/versioned/api/samza-sql.md]: This link is broken as you updated the examples. Could you fix it ?

Comment (shenodaguirguis) [reply docs/learn/documentation/versioned/api/samza-sql.md]: fixed

File: docs/learn/documentation/versioned/api/samza-sql.md [hunk 4]
```diff
@@ -19,65 +19,70 @@ title: Samza SQL
...
-
-```bash
-> ./deploy/samza/bin/run-app.sh --config-factory=org.apache.samza.config.factories.PropertiesConfigFactory --config-path=file://$PWD/deploy/samza/config/page-view-filter-sql.properties
-```
+ Operation | Syntax hints | Comments      
+ --- | --- | --- 
+ PROJECTION | SELECT/INSERT/UPSERT | See [SQL Grammar](#sql-grammer) below 
+ FILTERING | WHERE expression |See [SQL Grammar](#sql-grammer) below 
+ UDFs | udf_name(args)    | In both SELECT and WHERE clause 
+ JOIN | LEFT, RIGHT, FULL and NATURAL | Stream-table join. Currently not fully stable.
```
Comment (atoomula) [inline docs/learn/documentation/versioned/api/samza-sql.md]: Only Inner joins, Left and Right outer joins are supported. Cross joins, full outer joins or natural joins are not supported.

File: docs/learn/documentation/versioned/api/samza-sql.md [hunk 5]
```diff
@@ -19,65 +19,70 @@ title: Samza SQL
...
-```bash
-> ./deploy/samza/bin/run-app.sh --config-factory=org.apache.samza.config.factories.PropertiesConfigFactory --config-path=file://$PWD/deploy/samza/config/page-view-filter-sql.properties
-```
+ Operation | Syntax hints | Comments      
+ --- | --- | --- 
+ PROJECTION | SELECT/INSERT/UPSERT | See [SQL Grammar](#sql-grammer) below 
+ FILTERING | WHERE expression |See [SQL Grammar](#sql-grammer) below 
+ UDFs | udf_name(args)    | In both SELECT and WHERE clause 
+ JOIN | LEFT, RIGHT, FULL and NATURAL | Stream-table join. Currently not fully stable. 
+ AGGREGATION | GROUP BY | Currently not fully stable.
```
Comment (atoomula) [inline docs/learn/documentation/versioned/api/samza-sql.md]: Only COUNT aggregate function is supported and window function is based on processing time.

File: docs/learn/documentation/versioned/api/samza-sql.md (hunk 4 above)
Comment (shenodaguirguis) [reply docs/learn/documentation/versioned/api/samza-sql.md]: fixed. Thanks!

File: docs/learn/documentation/versioned/api/samza-sql.md (hunk 5 above)
Comment (shenodaguirguis) [reply docs/learn/documentation/versioned/api/samza-sql.md]: fixed

**Comment by vjagadish1989** [comment]
Thanks @atoomula and @shenodaguirguis . Maybe we already did this.. If not, would be nice to run through these setup instructions as a newbie user and verify they work as-is.

**Pull request closed without merging**